Sources/SecureXPC/Client/XPCClient.swift

```diff
@@ -123,20 +123,33 @@ public class XPCClient {
     ///    - named: A key in the `MachServices` entry of the tool's launchd property list.
     /// - Returns: A client configured to communicate with the named service.
     public static func forMachService(named machServiceName: String) -> XPCClient {
-        XPCMachClient(serviceName: machServiceName)
+        XPCMachClient(machServiceName: machServiceName)
+    }
+
+	public static func forEndpoint(_ endpoint: XPCServerEndpoint) -> XPCClient {
+        let connection = xpc_connection_create_from_endpoint(endpoint.endpoint)
+
+        xpc_connection_set_event_handler(connection, { (event: xpc_object_t) in
+            fatalError("Is should be impossible for this connection to recieve an event.")
+        })
+        xpc_connection_resume(connection)
```

Comment:
FIXME: we shouldn't resume the client here. We should probably defer it until `getConnection()`.